stlab/channel.hpp

```diff
@@ -434,6 +448,10 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
         bool do_run;
         {
             std::unique_lock<std::mutex> lock(_process_mutex);
+            while (_process_queue_size != 0 && _process_queue_size <= _process_message_queue.size())
+            {
+                _process_queue_condition.wait(lock);
```

Comment:
This was a misunderstanding. I understood, that it was your intention to limit the incoming queue. Will change it.